### ISSUE TITLE
Auto-update referrers when the page gets moved

### ISSUE THREAD

**Issue by vadosnaprimer** [adelikat/tasvideos#268]
Now it has to be done manually each time (thankfully we even have referrer lists), but it makes a lot of sense to update them automatically to point to the new page location.

**Comment by adelikat** [comment]
To be clear, you want it to update those wiki page's markup to have the new page name? That could be an expensive operation.

**Comment by vadosnaprimer** [comment]
I don't think it matters too much how the link is formatted, as long as it's updated (for example, TVA knows how to turn full wiki links into short ones).

**Comment by YoshiRulz** [comment]
IIRC MediaWiki does this by using an internal ID for pages for which the path acts as a pointer

**Comment by Randomno** [comment]
Doesn't MediaWiki just use page redirects?

**Comment by nattthebear** [comment]
This is sort of blocked on https://github.com/TASVideos/tasvideos/issues/219.  Doing this in a sane way would require a system that can both parse to ast and emit wiki syntax.

**Comment by vadosnaprimer** [comment]
It would be incredibly helpful to have this before we start heavily reorganizing the wiki pages. If something is linked to a lot and you have to move it, the routine of updating every referrer will get overwhelming very quickly.

**Comment by MemoryTAS** [comment]
Leaving Redirects like in mediawiki behind when moving would probably be optimal, lots of forum posts will no doubt be broken if I move say Judge Guidelines.

https://www.mediawiki.org/wiki/Help:Redirects
https://www.mediawiki.org/wiki/Manual:Redirects

**Comment by nattthebear** [comment]
We could auto update forum posts as well 🤔

**Comment by Randomno** [comment]
Any update on this?

Bare minimum needed is a redirect module that, well, redirects.

Other desired but unnecessary features:

* Notice of redirection
![image](https://github.com/TASVideos/tasvideos/assets/17330834/e5baef66-b2ed-4247-807d-a0f21cfb03d6)

* Way to view the page without redirection like https://en.wikipedia.org/w/index.php?title=UK&redirect=no

* Automatic creation of a redirect when moving a page. Without it, the former page will need to be re-created and have the redirect module added, but that's still far less effort than changing every existing link.
